static/js/search/all_results.tsx

```diff
@@ -82,32 +83,57 @@ export function AllResults(props: ResultsProps): JSX.Element {
   }, [props]);
 
   useEffect(() => {
-    getStatVarSearchResults(props.query, [])
-      .then((data) => {
-        const statVars: NamedNode[] = [];
-        data.statVarGroups.forEach((svg) => {
-          if (!_.isEmpty(svg.statVars)) {
-            statVars.push(...svg.statVars);
+    if (!props.selectedStatVar) {
+      getStatVarSearchResults(props.query, [])
+        .then((data) => {
+          const statVars: NamedNode[] = [];
+          data.statVarGroups.forEach((svg) => {
+            if (!_.isEmpty(svg.statVars)) {
+              statVars.push(...svg.statVars);
+            }
+          });
+          if (!_.isEmpty(data.statVars)) {
+            statVars.push(...data.statVars);
           }
+          setStatVars(statVars);
+        })
+        .catch(() => {
+          setStatVars([]);
         });
-        if (!_.isEmpty(data.statVars)) {
-          statVars.push(...data.statVars);
-        }
-        setStatVars(statVars);
-      })
-      .catch(() => {
-        setStatVars([]);
-      });
+    } else {
+      setStatVars([]);
+    }
   }, [props]);
 
   if (_.isNull(places) || _.isNull(statVars)) {
```

Comment:
could you add some comments as to the required states for these? null is pre-request and [] if there are no results. this would also be easier to read if it was "placeResults" and "statVarResults"

Reply:
done